core-bundle/src/Resources/contao/controllers/BackendPassword.php

```diff
@@ -52,25 +51,37 @@ public function __construct()
 	 */
 	public function run()
 	{
+		$container = System::getContainer();
+
 		/** @var Request $request */
-		$request = System::getContainer()->get('request_stack')->getCurrentRequest();
+		$request = $container->get('request_stack')->getCurrentRequest();
+
+		Controller::loadDataContainer('tl_user');
+
+		$dc = new DC_Table('tl_user');
+		$dc->id = $this->User->id;
+		$dc->activeRecord = $this->User;
+
+		$widget = new Password(Widget::getAttributesFromDca($GLOBALS['TL_DCA']['tl_user']['fields']['password'], 'password'));
+		$widget->template = 'be_widget_pwchange';
+		$widget->dataContainer = $dc;
+		$widget->confirmLabel = $container->get('translator')->trans('MSC.confirm.0', array(), 'contao_default');
 
 		$objTemplate = new BackendTemplate('be_password');
+		$objTemplate->widget = $widget->parse();
 
 		if (Input::post('FORM_SUBMIT') == 'tl_password')
 		{
+			$widget->validate();
+
+			// Remove any messages added by the Password widget itself
+			Message::reset();
```

Comment:
This seems strange to me. Why is there a _Please enter a password_ error message if a password has been provided?

Reply:
The _Pleaser enter a **new** password._ error occurs, if you enter the same password as you previously had again. The `BackendPassword` controller checks for this and displays this error message via `Message::addError`. However, the `Contao\Password` widget does not care about that and thus, since the password validates, wants to add confirmation message that the password has been updated via `Message::addConfirmation`.